### ISSUE TITLE
Dolt panics when renaming index containing capital letters

### ISSUE THREAD

**Issue by arvidfm** [dolthub/dolt#7945]
Using `ALTER TABLE ... RENAME INDEX` to rename an index causes Dolt to panic if the index name contains a capital letter.

MWE, works on Percona Server 8.3.0:
```sql
DROP TABLE IF EXISTS index_test;

CREATE TABLE `index_test` (
    `value` int NOT NULL,
    KEY myIndex (`value`)
);

ALTER TABLE index_test RENAME INDEX myIndex TO my_index;

DROP TABLE index_test;
```

<details>
<summary>Full stack trace</summary>

```
runtime error: invalid memory address or nil pointer dereference
go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.3.linux-amd64/src/runtime/panic.go:261 (0x4470fd)
        panicmem: panic(memoryError)
go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.3.linux-amd64/src/runtime/signal_unix.go:881 (0x461ce4)
        sigpanic: panicmem()
dolt/go/libraries/doltcore/schema/index_coll.go:455 (0x18ee159)
        com/dolthub/dolt/go/libraries/doltcore/schema.(*indexCollectionImpl).RenameIndex: index.name = newName
dolt/go/libraries/doltcore/sqle/tables.go:2374 (0x310ed0d)
        com/dolthub/dolt/go/libraries/doltcore/sqle.(*AlterableDoltTable).RenameIndex: _, err := t.sch.Indexes().RenameIndex(fromIndexName, toIndexName)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/sql/rowexec/ddl_iters.go:2022 (0x290e3e6)
        com/dolthub/go-mysql-server/sql/rowexec.(*BaseBuilder).executeAlterIndex: return idxAltTbl.RenameIndex(ctx, n.PreviousIndexName, n.IndexName)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/sql/rowexec/dml.go:283 (0x2916c98)
        com/dolthub/go-mysql-server/sql/rowexec.(*BaseBuilder).buildAlterIndex: err := b.executeAlterIndex(ctx, n)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/sql/rowexec/node_builder.gen.go:234 (0x29400c4)
        com/dolthub/go-mysql-server/sql/rowexec.(*BaseBuilder).buildNodeExecNoAnalyze: return b.buildAlterIndex(ctx, n, row)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/sql/rowexec/node_builder.gen.go:27 (0x293500f)
        com/dolthub/go-mysql-server/sql/rowexec.(*BaseBuilder).buildNodeExec: iter, err := b.buildNodeExecNoAnalyze(ctx, n, row)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/sql/rowexec/builder.go:33 (0x28df9e6)
        com/dolthub/go-mysql-server/sql/rowexec.(*BaseBuilder).Build: return b.buildNodeExec(ctx, n, r)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/sql/rowexec/transaction.go:305 (0x299aa84)
        com/dolthub/go-mysql-server/sql/rowexec.(*BaseBuilder).buildTransactionCommittingNode: iter, err := b.Build(ctx, n.Child(), row)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/sql/rowexec/node_builder.gen.go:48 (0x29360fd)
        com/dolthub/go-mysql-server/sql/rowexec.(*BaseBuilder).buildNodeExecNoAnalyze: return b.buildTransactionCommittingNode(ctx, n, row)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/sql/rowexec/node_builder.gen.go:27 (0x293500f)
        com/dolthub/go-mysql-server/sql/rowexec.(*BaseBuilder).buildNodeExec: iter, err := b.buildNodeExecNoAnalyze(ctx, n, row)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/sql/rowexec/builder.go:33 (0x28df9e6)
        com/dolthub/go-mysql-server/sql/rowexec.(*BaseBuilder).Build: return b.buildNodeExec(ctx, n, r)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/sql/rowexec/other.go:326 (0x294a3e4)
        com/dolthub/go-mysql-server/sql/rowexec.(*BaseBuilder).buildQueryProcess: iter, err := b.Build(ctx, n.Child(), row)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/sql/rowexec/node_builder.gen.go:58 (0x29449a4)
        com/dolthub/go-mysql-server/sql/rowexec.(*BaseBuilder).buildNodeExecNoAnalyze: return b.buildQueryProcess(ctx, n, row)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/sql/rowexec/node_builder.gen.go:27 (0x293500f)
        com/dolthub/go-mysql-server/sql/rowexec.(*BaseBuilder).buildNodeExec: iter, err := b.buildNodeExecNoAnalyze(ctx, n, row)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/sql/rowexec/builder.go:33 (0x28df9e6)
        com/dolthub/go-mysql-server/sql/rowexec.(*BaseBuilder).Build: return b.buildNodeExec(ctx, n, r)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/engine.go:408 (0x2b7082a)
        com/dolthub/go-mysql-server.(*Engine).QueryWithBindings: iter, err := e.Analyzer.ExecBuilder.Build(ctx, analyzed, nil)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/server/handler.go:961 (0x2f32c6d)
        com/dolthub/go-mysql-server/server.(*Handler).executeQuery: return h.e.QueryWithBindings(ctx, query, parsed, bindings)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/server/handler.go:409 (0x2f2c66e)
        com/dolthub/go-mysql-server/server.(*Handler).doQuery: schema, rowIter, err := queryExec(ctx, query, parsed, analyzedPlan, bindings)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/server/handler.go:665 (0x2f2fe66)
        com/dolthub/go-mysql-server/server.(*Handler).errorWrappedDoQuery: remainder, err := h.doQuery(c, query, parsed, nil, mode, h.executeQuery, bindings, callback)
go/pkg/mod/github.com/dolthub/go-mysql-server@v0.18.2-0.20240529234304-47f89331b0ea/server/handler.go:283 (0x2f2b214)
        com/dolthub/go-mysql-server/server.(*Handler).ComMultiQuery: return h.errorWrappedDoQuery(c, query, nil, MultiStmtModeOn, nil, callback)
go/pkg/mod/github.com/dolthub/vitess@v0.0.0-20240524204318-4c419c3de9c4/go/mysql/conn.go:1548 (0xd02781)
        com/dolthub/vitess/go/mysql.(*Conn).execQuery: remainder, err = handler.ComMultiQuery(c, query, resultsCB)
go/pkg/mod/github.com/dolthub/vitess@v0.0.0-20240524204318-4c419c3de9c4/go/mysql/conn.go:991 (0xcfa5f8)
        com/dolthub/vitess/go/mysql.(*Conn).handleNextCommand: for query, err = c.execQuery(query, handler, multiStatements); err == nil && query != ""; {
go/pkg/mod/github.com/dolthub/vitess@v0.0.0-20240524204318-4c419c3de9c4/go/mysql/server.go:535 (0xd35610)
        com/dolthub/vitess/go/mysql.(*Listener).handle: err := c.handleNextCommand(l.handler)
go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.3.linux-amd64/src/runtime/asm_amd64.s:1695 (0x4845a0)
        goexit: BYTE    $0x90   // NOP 
```
</details>

**Comment by fulghum** [comment]
Hi @arvidfm, thanks for taking the time to report this one. I'll get started debugging through the repro and let you know what I find.

**Comment by fulghum** [comment]
The root cause of this one was a lack of handling index names case-insensitively. We've added more tests to cover this and changed the index logic to work with names in a case-insensitive manner. 

We're cleaning up our CI pipeline, then we'll get the PRs merged in, and fire off a release for you.

**Comment by fulghum** [comment]
@arvidfm – The fix for this issue has been released in [Dolt 1.39.3](https://github.com/dolthub/dolt/releases/tag/v1.39.3). Let us know if you hit any other snags and thanks again for helping us find this issue! 🙏

**Comment by arvidfm** [comment]
Amazing, thank you so much for the quick response! I know there was at least one other issue I found regarding case sensitivity (though it doesn't cause a panic), I'll check if it's still a problem with this latest version, and if not I'll create an issue once I've put together a MWE.
